lib/modules/platform/gitea/index.ts

```diff
@@ -183,7 +183,9 @@ const platform: Platform = {
     }
 
     if (endpoint) {
-      defaults.endpoint = ensureTrailingSlash(endpoint);
+      let baseEndpoint = trimTrailingApiPath(endpoint);
+      baseEndpoint = ensureTrailingSlash(baseEndpoint);
+      defaults.endpoint = baseEndpoint;
     } else {
       logger.debug('Using default Gitea endpoint: ' + defaults.endpoint);
```

Comment:
does the default endpoint also doesn't have the API ?

Reply:
The default.endpoint does not contain the api path name `api/v1`. For the api calls, the path name is appended in the gitea helper. `trimTrailingApiPath()` ensures that an optionally specified endpoint environment variable containing the api path name is trimmed.

Reply:
it has, please update L55 and remove api
https://github.com/renovatebot/renovate/blob/dca3418bbdbf1024ac6807e591562b4042bc846d/lib/modules/platform/gitea/index.ts#L55

Please add a init repo test without passing a endpoint

Reply:
Why would such a test be necessary? `initRepo()` requests the repo properties via `helper.getRepo()`, that is mocked in the index.spec.ts tests. Afterwards, the [repo.clone_url](https://github.com/dome4/renovate/blob/8d1d21b1e7349b9eb98fdcef2c6a8408422ce110/lib/modules/platform/gitea/index.ts#L303) is used for `git.initRepo()`. In my opinion, the endpoint setting needs to be tested for `initPlatform()`. There are two tests for the default endpoint and a custom endpoint, both of which are tested via snapshot tests. I haven't yet understood what you think is missing.

Reply:
you are right, i liked to test init platform instead of init repo. i see now that you've updated the tests.